reana_job_controller/compute4punch_job_manager.py

```diff
@@ -0,0 +1,373 @@
+# This file is part of REANA.
+# Copyright (C) 2019, 2020, 2021, 2022, 2023, 2024 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Compute4PUNCH Job Manager."""
+
+import base64
+import logging
+import os
+import re
+
+from paramiko.sftp_client import SFTPClient
+from stat import S_ISDIR
+from typing import Iterable
+
+from reana_commons.workspace import is_directory, open_file, walk
+from reana_job_controller.job_manager import JobManager
+from reana_job_controller.utils import SSHClient, motley_cue_auth_strategy_factory
+from reana_job_controller.config import (
+    C4P_LOGIN_NODE_HOSTNAME,
+    C4P_LOGIN_NODE_PORT,
+    C4P_SSH_TIMEOUT,
+    C4P_SSH_BANNER_TIMEOUT,
+    C4P_SSH_AUTH_TIMEOUT,
+    C4P_CPU_CORES,
+    C4P_MEMORY_LIMIT,
+    C4P_ADDITIONAL_REQUIREMENTS,
+    C4P_REANA_REL_WORKFLOW_PATH,
+)
+
+
+class Compute4PUNCHJobManager(JobManager):
+    """Compute4PUNCH Job Manager."""
+
+    C4P_WORKSPACE_PATH = ""
+    """Absolute path on the Compute4PUNCH head node used for submission"""
+    C4P_HOME_PATH = ""
+    """Default Compute4PUNCH home directory"""
+    SUBMIT_ID_PATTERN = re.compile(r"Proc\s(\d+\.\d+)")
+    """ regex to search the Job ID in a submit Proc line """
+
+    def __init__(
+        self,
+        docker_img=None,
+        cmd=None,
+        prettified_cmd=None,
+        env_vars=None,
+        workflow_uuid=None,
+        workflow_workspace=None,
+        cvmfs_mounts="false",
+        shared_file_system=False,
+        job_name=None,
+        c4p_cpu_cores=C4P_CPU_CORES,
+        c4p_memory_limit=C4P_MEMORY_LIMIT,
+        c4p_additional_requirements=C4P_ADDITIONAL_REQUIREMENTS,
```

Comment:
One more question: are these value supposed to be user configurable? If so, would you like to do it now or at a later stage?

Reply:
Yes, these values are supposed to be user configurable. I think we potentially should first agree upon a common syntax for all job controllers as mentioned in https://github.com/reanahub/reana-job-controller/pull/430#pullrequestreview-1901433543, right?

Reply:
We can harmonise them in a later release, so we can proceed with these values. Do you already have changed all the r-w-engine-* to load these values from the specifications?

Reply:
Not yet, because of thinking using something harmonised. I will have look into it .

Reply:
No problem, I can also help with this if needed, just let me know